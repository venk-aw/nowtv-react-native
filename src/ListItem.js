import React, {Component} from 'react'
import {View, Text, StyleSheet, Image, TouchableWithoutFeedback} from 'react-native'

const styles = StyleSheet.create({
  container: {
    paddingVertical: 4,
    paddingHorizontal: 8,
    borderBottomWidth: 1,
    minHeight: 48,
    flexDirection: 'row',
    alignItems: 'center'
  },
  emailPlaceHolder: {
    minHeight: 18
  },
  avatar: {
    width: 48,
    height: 48,
    borderRadius: 24,
    borderWidth: 1,
    borderColor: '#dddddd',
    overflow: 'hidden',
    justifyContent: 'center',
    alignItems: 'center'
  },
  textBlock: {
    paddingLeft: 4
  },
  image: {
    width: 46,
    height: 46
  }
})

export default class ListItem extends Component {
  state = {
    showUserDetails: false
  }

  onMessageSelect = () => {
    this.setState({
      showUserDetails: true,
    });
  }

  hideUserDetails = () => {
    this.setState({
      showUserDetails: false
    });
  }

  render () {
    const {message, user} = this.props
    const {showUserDetails} = this.state
    return (
      <TouchableWithoutFeedback onPressIn={this.onMessageSelect} onPressOut={this.hideUserDetails} delayPressOut={1500}>
        <View style={styles.container}>
          <View style={styles.avatar}>
            {user && user.avatar && <Image source={{uri: user.avatar}} style={styles.image}/>}
          </View>
          <View style={styles.textBlock}>
            <Text>{message}</Text>
            <View style={styles.emailPlaceHolder}>
              {showUserDetails &&
                <Text>{user.email}</Text>
              }
            </View>
          </View>
        </View>
      </TouchableWithoutFeedback>
    )
  }
}