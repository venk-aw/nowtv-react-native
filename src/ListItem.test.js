import React from 'react';
import ListItem from './ListItem';
import renderer from 'react-test-renderer';

describe('ListItem', () => {
  it ('should render without exploding', () => {
    const tree = renderer.create(<ListItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it ('should render without failure even if no user is passed', () => {
    const minProps = {
      id: 'cd445e6d-e514-424f-ba8f-16ec842004da',
      userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
      message: 'Lorem ipsum',
      timestamp: '2017-01-11T01:07:00Z'
    };
    const tree = renderer.create(<ListItem {...minProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it ('should render the image', () => {
    const minProps = {
      id: 'cd445e6d-e514-424f-ba8f-16ec842004da',
      userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
      message: 'Lorem ipsum',
      timestamp: '2017-01-11T01:07:00Z',
      user: {
        "id": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
        "firstName": "Martin",
        "lastName": "Bradley",
        "email": "mbradley0@google.it",
        "avatar": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        "ip": "166.124.172.160"
      }
    };
    const tree = renderer.create(<ListItem {...minProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it ('should not render the image if no URL provided', () => {
    const minProps = {
      id: 'cd445e6d-e514-424f-ba8f-16ec842004da',
      userId: 'fe27b760-a915-475c-80bb-7cdf14cc6ef3',
      message: 'Lorem ipsum',
      timestamp: '2017-01-11T01:07:00Z',
      user: {
        "id": "fe27b760-a915-475c-80bb-7cdf14cc6ef3",
        "firstName": "Martin",
        "lastName": "Bradley",
        "email": "mbradley0@google.it",
        "avatar": null,
        "ip": "166.124.172.160"
      }
    };
    const tree = renderer.create(<ListItem {...minProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
