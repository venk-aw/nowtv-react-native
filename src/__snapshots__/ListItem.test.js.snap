// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should not render the image if no URL provided 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "minHeight": 48,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#dddddd",
        "borderRadius": 24,
        "borderWidth": 1,
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Lorem ipsum
    </Text>
    <View
      style={
        Object {
          "minHeight": 18,
        }
      }
    />
  </View>
</View>
`;

exports[`ListItem should render the image 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "minHeight": 48,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#dddddd",
        "borderRadius": 24,
        "borderWidth": 1,
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
        }
      }
      style={
        Object {
          "height": 46,
          "width": 46,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "paddingLeft": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Lorem ipsum
    </Text>
    <View
      style={
        Object {
          "minHeight": 18,
        }
      }
    />
  </View>
</View>
`;

exports[`ListItem should render without exploding 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "minHeight": 48,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#dddddd",
        "borderRadius": 24,
        "borderWidth": 1,
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <View
      style={
        Object {
          "minHeight": 18,
        }
      }
    />
  </View>
</View>
`;

exports[`ListItem should render without failure even if no user is passed 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "minHeight": 48,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#dddddd",
        "borderRadius": 24,
        "borderWidth": 1,
        "height": 48,
        "justifyContent": "center",
        "overflow": "hidden",
        "width": 48,
      }
    }
  />
  <View
    style={
      Object {
        "paddingLeft": 4,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Lorem ipsum
    </Text>
    <View
      style={
        Object {
          "minHeight": 18,
        }
      }
    />
  </View>
</View>
`;
